### PR TITLE
Add torch-pme recipe

### PR DESCRIPTION
**Pull request by PicoCentauri**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Luthaf** [comment]
I'm happy to help maintain this package!

**Comment by PicoCentauri** [comment]
@conda-forge/help-python, ready for review!

File: recipes/torch-pme/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{% set name = "torch_pme" %}
+{% set version = "0.1.0" %}
+
+package:
+  # use name with a hyphen to be consistent with PyPI name
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:5]:
```suggestion
```

File: recipes/torch-pme/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{% set name = "torch_pme" %}
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:1]: This is the name and should be the same as PyPI.

```suggestion
{% set name = "torch-pme" %}
```

File: recipes/torch-pme/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+{% set version = "0.1.0" %}
+
+package:
+  # use name with a hyphen to be consistent with PyPI name
+  name: torch-pme
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:10]:
```suggestion
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/torch_pme-{{ version }}.tar.gz
```

File: recipes/torch-pme/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 7d31801b44a19a1633ecc4f2c484811342940b074dcd01f30dd97ff5a4e69278
+
+build:
+  # conda-forge's pytorch is not yet available on Windows
+  skip: true  # [win or py<39]
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:15]: It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows. 

```suggestion
  noarch: python
```

File: recipes/torch-pme/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:22]:
```suggestion
    - python {{ python_min }}
```

File: recipes/torch-pme/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - setuptools
+    - wheel
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:26]:
```suggestion
    - python >={{ python_min }}
```

File: recipes/torch-pme/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    - pytorch >=2.3
+
+test:
+  imports:
+    - torchpme
+  commands:
+    - pip check
+  requires:
+    - pip
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:35]:
```suggestion
  requires:
    - python {{ python_min }}
    - pip
```

**Review by ocefpaf — COMMENTED** [review]
Please use grayskull to generate Python recipes. Grayskull make it easier for us to review.

See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi for more information.

**Comment by PicoCentauri** [comment]
@ocefpaf thanks for the comments and the suggestions of Grayskull. I was not aware that this tool exists. This will make creation much easier in the future. I hope I addressed all suggestions.

File: recipes/torch-pme/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/torch_pme/torch_pme-{{ version }}.tar.gz  
+  sha256: 7d31801b44a19a1633ecc4f2c484811342940b074dcd01f30dd97ff5a4e69278
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/torch-pme/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
+{% set name = "torch-pme" %}
+{% set version = "0.1.0" %}
+{% set python_min = "3.9" %}
```
Comment (ocefpaf) [inline recipes/torch-pme/meta.yaml:3]:
```suggestion
```

File: recipes/torch-pme/meta.yaml (hunk 7 above)
Comment (ocefpaf) [reply recipes/torch-pme/meta.yaml:34]: Ping

**Comment by ocefpaf** [comment]
Please don't request another review until you acted on the past one. You still need to add the python min to the test area as per my code suggestion from [before](https://github.com/conda-forge/staged-recipes/pull/28488#discussion_r1876284621)

**Comment by PicoCentauri** [comment]
Sorry, I thought this was not necessary because it is not explicitly required by the tests. I updated the branch.

**Comment by ocefpaf** [comment]
> Sorry, I thought this was not necessary because it is not explicitly required by the tests. I updated the branch.

It was in the linter notes, in your review, and I pinged you 2 times. Please trust the reviewers, that will save us a lot of our volunteered time.

**Comment by PicoCentauri** [comment]
Yes sorry again and thanks for the time. Really appreciate the work.

**Pull request merged**